services/website/website.service.js

```diff
@@ -38,118 +30,76 @@ const documentation = `
-      </td><td>  →
-      </td><td>  <code>-</code> Dash
-    </td></tr>
-    <tr><td>   Underscores <code>__</code>
-      </td><td>  →
-      </td><td>  <code>_</code> Underscore <br/>
-    </td></tr>
-    <tr><td>   Slashes <code>//</code>
-      </td><td>  →
-      </td><td>  <code>/</code> Slash <br/>
-    </td></tr>
-    <tr><td>   <code>_</code> or Space <code>&nbsp;</code>
-      </td><td>  →
-      </td><td>  <code>&nbsp;</code> Space
-    </td></tr>
-  </tbody></table>
+  The messages and colors for the up and down states can also be customized.
 </p>
 `
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class Website extends LegacyService {
+module.exports = class Website extends BaseService {
   static get category() {
     return 'monitoring'
   }
 
   static get route() {
     return {
-      base: '',
-      pattern: '',
+      base: 'website',
+      pattern: ':protocol(https|http)/:hostAndPath',
+      queryParamSchema,
     }
   }
 
   static get examples() {
     return [
       {
         title: 'Website',
-        pattern: 'website-up-down-green-red/:protocol(https|http)/:hostAndPath',
         namedParams: {
           protocol: 'https',
           hostAndPath: 'shields.io',
         },
-        staticPreview: {
-          label: 'website',
-          message: 'up',
-          color: 'green',
-        },
+        queryParams: exampleQueryParams,
+        staticPreview: renderWebsiteStatus({ isUp: true }),
         documentation,
       },
     ]
   }
 
-  static registerLegacyRouteHandler({ camp, cache }) {
-    camp.route(
-      /^\/website(-(([^-/]|--|\/\/)+)-(([^-/]|--|\/\/)+)(-(([^-/]|--|\/\/)+)-(([^-/]|--|\/\/)+))?)?\/([^/]+)\/(.+)\.(svg|png|gif|jpg|json)$/,
-      cache((data, match, sendBadge, request) => {
-        const onlineMessage = escapeFormatSlashes(
-          match[2] != null ? match[2] : 'online'
-        )
-        const offlineMessage = escapeFormatSlashes(
-          match[4] != null ? match[4] : 'offline'
-        )
-        const onlineColor = escapeFormatSlashes(
-          match[7] != null ? match[7] : 'brightgreen'
-        )
-        const offlineColor = escapeFormatSlashes(
-          match[9] != null ? match[9] : 'red'
-        )
-        const userProtocol = match[11]
-        const userURI = match[12]
-        const format = match[13]
-        const withProtocolURI = `${userProtocol}://${userURI}`
-        const options = {
+  static get defaultBadgeData() {
+    return {
+      label: 'website',
+    }
+  }
+
+  async handle(
+    { protocol, hostAndPath },
+    {
+      up_message: upMessage,
+      down_message: downMessage,
+      up_color: upColor,
+      down_color: downColor,
+    }
+  ) {
+    let isUp
+    try {
+      const {
+        res: { statusCode },
+      } = await this._request({
```

Comment:
`this._requestFetcher` throws http errors, so this try block is needed regardless.